### PR TITLE
API and DB

### PR DESCRIPTION
**Pull request by bernardlau**
- includes mySQL backend for users and audienceReviews
- includes express/node for API/server
      1. /reviews/audience/:title to retrieve four audience reviews
      2. /reviews/scoreboard/:title to retrieve number of reviews, stars, and % liked for a movie
- includes test cases for backend using Jest
- includes test cases for API using supertest and Jest
- includes audience review component using React

### CODE REVIEW

**Review by chenlwilson — COMMENTED** [review]
b013a70
You might want to .gitignore the node_modules
Just saw the next commit, scratch above...

File: data/scripts/generateMovieReviews.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        movieReview.createdAt = "";
+        movieReview.not_interested = 0;
+        movieReview.want_to_see_it = 0;
+        movieReview.liked = getRandomInt(2);
+        counter++;
+        i++;
+        movieData.push(movieReview);
+      }
+    });
+    fs.writeFile('data/movieReviews.json', JSON.stringify(movieData, null, 2), 'utf-8', (err) => {
```
Comment (chrisweilacker) [inline data/scripts/generateMovieReviews.js:43]: Shouldnt you being saving this to a database not a json file?

File: data/scripts/generateUsers.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  var user = {};
+  user.id = i;
+  user.username = loremArr[i];
+  user.has_profile_pic = getProfilePic();
+  user.etag = "";
+  user.objectURL = "";
+  userData.push(user);
+}
+
+fs.writeFile('data/user.json', JSON.stringify(userData, null, 2), 'utf-8', (err) => {
```
Comment (chrisweilacker) [inline data/scripts/generateUsers.js:24]: same thing do you tranfer this json file to your database at some point?

**Review by chrisweilacker — COMMENTED** [review]
As the nodemodules wasnt gitignored was very hard to read inside here so I read the code inside VSCode:

1.  I see your seed script to insert from the JSON.
2. App.js the config file doesnt seem to be there you shouldnt use the seed script had to help someone else who did that can cause massive errors in testing so please connect the config file.
3. App.js think you should refactor the database queries into its own file with helper functions probably can be inside the config.js database functions.
4. For the scoreboard user reviews service you can offload alot of this to the database to calculate rather than for looping in your app by using average, count and a few other built in database functions with some small amount of math you can take a look at my db file for some inspiration as my tomato-meter scoreboard should be pretty similar.

5. I dont have time now to analyze it completely but your tests seem a bit different in the way that it accesses the server so I will just trust that it works for now.

File: data/scripts/generateUsers.js (hunk 2 above)
Comment (bernardlau) [reply data/scripts/generateUsers.js:24]: I put all the data into a JSON first then put it into my db.

**Comment by bernardlau** [comment]
@chrisweilacker 
1. n/a
2. made config.js file public, figured this out when I had to run TravisCI, created env_variables to resolve this.
3. Refactored the db queries into their own helper function
4. I offloaded the audience ratings to the database side instead of the server
5. TravisCI is now working to show the passing test cases, see above check

**Pull request merged**